resources/builder.inc.sh

```diff
@@ -1327,18 +1327,21 @@ _builder_parse_expanded_parameters() {
     builder_echo setmark "dependency build, started by $builder_dep_parent"
     builder_echo grey "build.sh parameters: <${_params[@]}>"
     if [[ -z ${_builder_deps_built+x} ]]; then
-      builder_die "FATAL ERROR: Expected --builder-deps-built parameter"
+      builder_die "FATAL ERROR: Expected '_builder_deps_built' variable to be set"
+    fi
+  elif builder_is_child_build; then
+    builder_echo setmark "child build, parameters: <${_params[@]}>"
+    if [[ -z ${_builder_deps_built+x} ]]; then
+      builder_die "FATAL ERROR: Expected '_builder_deps_built' variable to be set"
     fi
   else
-    # This is a top-level invocation, not a dependency build, so we want to
-    # track which dependencies have been built, so they don't get built multiple
-    # times.
-    # TODO: consider printing expanded builder parameters instead of shorthand
+    # This is a top-level invocation, so we want to track which dependencies
+    # have been built, so they don't get built multiple times.
     builder_echo setmark "build.sh parameters: <${_params[@]}>"
     if [[ ${#builder_extra_params[@]} -gt 0 ]]; then
       builder_echo grey "build.sh extra parameters: <${builder_extra_params[@]}>"
     fi
-    _builder_deps_built=`mktemp`
+    export _builder_deps_built=`mktemp`
```

Comment:
I take it this is the main magic line, with the other changes being included to insure consistency with child scripts as well?

Reply:
Yep